clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -788,11 +788,13 @@ protected synchronized void markCoordinatorUnknown(boolean isDisconnected) {
     }
 
     /**
-     * Get the current generation state if the group is stable.
-     * @return the current generation or null if the group is unjoined/rebalancing
+     * Get the current generation state
+     *
+     * @param onlyIfStable if set to true, only return null when the group is stable.
```

Comment:
Can you explain this a bit further? Why do we return null when the group is stable, I assume stable means everyone is successfully on the same generation? So why return null rather than the actual generation.

Reply:
Previously `generation()` would return null if we are in the middle of the rebalance, which is not useful in our case (in line 582 of ConsumerCoordinator, we are indeed in the middle of a rebalance, so `generaiton()` would always return null).

Reply:
I see. Can you comment on what happens when `onlyIfStable` is false? Do we always return null in that case, or just return the last valid generation, or ...?

Reply:
I think you should have a separate method `generationIfStable` or something.

Reply:
Yeah I think that works too, previously I wanted to consolidate the logic in a single function, but just doing that in a different function may be clearer. But I'd need to renaming the existing `generation` to `generationIfStable` then.